napari/layers/_vectors_layer/view/properties.py

```diff
@@ -0,0 +1,125 @@
+
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QLabel, QComboBox, QDoubleSpinBox, QSpinBox, QGridLayout
+import numpy as np
+import scipy.signal as signal
+
+from ..._base_layer import QtLayer
+
+
+class QtVectorsLayer(QtLayer):
+
+    def __init__(self, layer):
+        super().__init__(layer)
+
+        self.layer.events.emit_avg.connect(self._default_avg)
+        self.layer.events.emit_len.connect(self._default_length)
+
+        # vector color adjustment and widget
+        face_comboBox = QComboBox()
+        colors = self.layer._colors
+        for c in colors:
+            face_comboBox.addItem(c)
+        index = face_comboBox.findText(self.layer.color, Qt.MatchFixedString)
+        if index >= 0:
+            face_comboBox.setCurrentIndex(index)
+        face_comboBox.activated[str].connect(
+            lambda text=face_comboBox: self.change_face_color(text))
+        self.grid_layout.addWidget(QLabel('color:'), 3, 0)
+        self.grid_layout.addWidget(face_comboBox, 3, 1)
+
+        # line width in pixels
+        width_field = QSpinBox()
+        value = self.layer.width
+        width_field.setValue(value)
+        width_field.valueChanged.connect(self.change_width)
+        self.grid_layout.addWidget(QLabel('width:'), 4, 0)
+        self.grid_layout.addWidget(width_field, 4, 1)
+
+        # averaging spinbox
+        self.averaging_spinbox = QSpinBox()
+        self.averaging_spinbox.setSingleStep(2)
+        self.averaging_spinbox.setValue(1)
+        self.averaging_spinbox.valueChanged.connect(self.change_average_type)
+        self.grid_layout.addWidget(QLabel('avg kernel'), 5, 0)
+        self.grid_layout.addWidget(self.averaging_spinbox, 5, 1)
+
+        # line length
+        self.length_field = QDoubleSpinBox()
+        self.length_field.setSingleStep(0.1)
+        value = self.layer.length
+        self.length_field.setValue(value)
+        self.length_field.valueChanged.connect(self.change_length)
+        self.grid_layout.addWidget(QLabel('length:'), 6, 0)
+        self.grid_layout.addWidget(self.length_field, 6, 1)
+
+        self.setExpanded(False)
+
+    def change_face_color(self, text):
+        self.layer.color = text
+
+    def change_connector_type(self, text):
+        self.layer.connector = text
+
+    def change_average_type(self, value):
+        self.layer.averaging = value
+
+    def change_width(self, value):
+        self.layer.width = value
+    
+    def change_length(self, value):
+        self.layer.length = value
+
+    def _default_avg(self, event_kernel):
```

Comment:
My thoughts are that all this averaging should not live in the properties file, but in the model file. It is more consistent with the rest of the layers to do it this way. Curious what @royerloic thinks

Reply:
I've changed it now.  I placed it there because I saw it as a slot and wanted to stay consistent with where slots and signals existed.  But it's simple to make another properties file method that calls the layer._default_avg -- so I've done this

Reply:
there's another concern I have here, which might be fixable:

as it exists, the flow looks like this:
1) user changes averaging in UI properties
2) layer.averaging property is directly called and set
3) layer.averaging property setter emits a signal
4) UI receives the signal and calls another UI method
5) this other UI method calls layer._default_avg

So now we have a situation where the calls bounce back and forth between layer and properties.  But it is necessitated by the fact that I want observers for this UI event, and I want to silo where signals and slots exist in a manner consistent with how properties/layers are written.

Reply:
yup, I think we will likely revisit this when we refactor our layers (like how we are refactoring our dims), but for now it is consistent with the other layer types. Can you want to post this concern as an issue that we can then discuss more.